### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Chincoya**
Add RSpec Tests.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      # 9 check - 21 in total
+      if x == 7
+        it 'Check if tied' do
+          engine1.move(x)
+          expect(engine1.game_end?).to be_falsy
+        end
+      else
+        it "Checks win condition for #{x}" do
+          engine1.move(x)
+          expect(engine1.game_end?).to eql(-1)
```
Comment (whiz25) [inline spec/spec_helper.rb]: TicTacLogic#game_end Checks win condition for 3
     Failure/Error: expect(engine1.game_end?).to eql(-1)
     
       expected: -1
            got: true
     
       (compared using eql?)
     # ./spec/spec_helper.rb:38:in `block (4 levels) in <top (required)>'

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  end
+
+  describe '#game_end' do
+    engine1 = TicTacLogic.new('P1', 'P2')
+    [1, 2, 3, 4, 6, 5, 8, 9, 7].each do |x|
+      # 9 check - 21 in total
+      if x == 7
+        it 'Check if tied' do
+          engine1.move(x)
+          expect(engine1.game_end?).to be_falsy
```
Comment (whiz25) [inline spec/spec_helper.rb]: TicTacLogic#game_end Check if tied
     Failure/Error: expect(engine1.game_end?).to be_falsy
     
       expected: falsey value
            got: true
     # ./spec/spec_helper.rb:33:in `block (4 levels) in <top (required)>'

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    end
+
+    engine8 = TicTacLogic.new('P1', 'P2')
+    [1, 2, 5, 3, 9].each do |x|
+      engine7.move(x)
+      # 1 check - 28 in total
+      next unless x == 9
+
+      it 'Checks win condition on first diagonal' do
+        expect(engine8.game_end?).to be_truthy
```
Comment (whiz25) [inline spec/spec_helper.rb]: TicTacLogic#game_end Checks win condition on first diagonal
      Failure/Error: elsif @board.none?(Integer)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./lib/logic.rb:26:in `none?'
      # ./lib/logic.rb:26:in `game_end?'
      # ./spec/spec_helper.rb:116:in `block (4 levels) in <top (required)>'

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  end
+
+  describe '#current_player' do
+    engine10 = TicTacLogic.new('P1', 'P2')
+    # 2 checks - 31 in total
+    it 'Checks if the current player is the first one' do
+      expect(engine10.current_player).to eql('P1')
+    end
+    it 'Checks if the current player is the second one' do
+      engine10.game_end?
```
Comment (whiz25) [inline spec/spec_helper.rb]: TicTacLogic#current_player Checks if the current player is the second one
      Failure/Error: elsif @board.none?(Integer)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./lib/logic.rb:26:in `none?'
      # ./lib/logic.rb:26:in `game_end?'
      # ./spec/spec_helper.rb:139:in `block (3 levels) in <top (required)>'

File: spec/spec_helper.rb (hunk 1 above)
Comment (Chincoya) [reply spec/spec_helper.rb]: image

File: spec/spec_helper.rb (hunk 2 above)
Comment (Chincoya) [reply spec/spec_helper.rb]: image

File: spec/spec_helper.rb (hunk 3 above)
Comment (Chincoya) [reply spec/spec_helper.rb]: image

File: spec/spec_helper.rb (hunk 4 above)
Comment (Chincoya) [reply spec/spec_helper.rb]: image

**Pull request merged**